### DISCUSSION TITLE
Using DaxStudio and RLS

### DISCUSSION THREAD

**Discussion by erapade** [DaxStudio/DaxStudio#710] [Q&A] [answered]
I guess I figured out what I need to do to enable a Role from DaxStudio, but I didn't manage to change the effective user, so I wonder if there are any instructions for this. 

One thing I noticed was that it looked like there was more queries running the first time. As I understand, this is to sort out what data I have access to and this is then saved in some temporary table. I could then see this code almost lite below executed:
SELECT XXX FROM FACT
LEFT JOIN 'RLS Table' On xxxx
WHERE VERTICALC ( 'RLS_Table'.$ROWFILTER IN '0x0000000000000013' )

I guess what I'm hinting here is a Blog post subject for Marco/Alberto ;-)

**Comment by marcosqlbi** [comment, marked as answer]
Fresh article from 2016 :-)
https://www.sqlbi.com/articles/security-cost-in-analysis-services-tabular/

**Comment by dgosbell** [comment]
>  I didn't manage to change the effective user, so I wonder if there are any instructions for this.

Can you provide more information about this? What sort of connection were you using? Did you get an error? If so what did it say?

I'm pretty sure Microsoft only supports impersonating Windows accounts using the EffectiveUsername property. So if you are trying to connect to the XMLA endpoint or ASAzure it might not work. I've been looking at implementing a similar "View As" functionality to the one in Power BI, but it uses undocumented features so it's not a straight forward process.

**Reply by erapade** [reply]
@dgosbell: I'm using DaxStudio as ExternalTool directly from PowerBI Desktop, so it's a Tabular Server connection to local host and port number. In Advanced settings I'm using defaults for DQ mode, MDX Comp (3), and Locale. So the only thing I have changed is the Role, and this part works fine. But when I then Specifying a user account using the form DOMAIN\userid I get this error, even though I'm using the same user ID as specified in the top window frame of DaxStudio
![image](https://user-images.githubusercontent.com/17324892/133038516-c576e27f-09de-4eca-92f0-a7e99b3dbc38.png)
